src/components/ShareResultsModal/EmailField.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import * as z from 'zod';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+import CheckMarkCircleSVG from './CheckMarkCircleSVG';
+
+const emailSchema = z.object({
+  email: z.string().email({ message: 'Please enter a valid email address' }),
+});
+
+export default function EmailField({ onSubmit }: Props) {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitSuccessful },
+  } = useForm<{ email: string }>({
+    resolver: zodResolver(emailSchema),
+  });
+
+  return (
+    <div className='cio-share-results-feature-group'>
+      <form onSubmit={handleSubmit(({ email }) => onSubmit(email))}>
+        <div className='cio-share-results-description'>Share by email</div>
+        <div className='cio-share-results-button-group'>
+          <div className='cio-share-results-email-input-group'>
+            <input
+              className={`cio-share-results-email-input ${
+                errors.email ? 'cio-share-results-email-input--error' : ''
+              }`}
+              {...register('email')}
+            />
+            {errors.email && (
+              <div className='cio-share-results-email-input-error-message'>
+                {errors.email.message}
+              </div>
+            )}
+          </div>
+          <button className='cio-share-results-share-button' type='submit'>
+            Send
+          </button>
+        </div>
+      </form>
+      {isSubmitSuccessful && (
+        <div className='cio-share-results-copied-notification'>
+          <CheckMarkCircleSVG />
+          <div>Email sent</div>
```

Comment:
I can imagine this becoming a problem since it indicates that the email has been sent when it actually means that the form has been submitted. Not sure how to handle this though

Reply:
Yeah this is a weird one :x I think what we can do to make this better is to make email submit callback async and only show this message if the callback successfully resolves

Reply:
ab384b91031ea7727d4bf0bda8d162af1a9b9e6e handleSubmit from react-hook-forms takes an async `onSubmit` function and only changes state to `isSubmitSuccessful` once the promise resolves

Reply:
Nice 🧠